omiclearn/utils/ui_components.py

```diff
@@ -608,45 +608,79 @@ def _generate_eda_section(state):
 # Generate additional feature selection section
 def _generate_additional_feature_selection_section(state, multiselect):
     with st.expander("Additional features"):
-        st.markdown(
-            "Select additional features. All non numerical values will be encoded (e.g. M/F -> 0,1)"
+        st.markdown(ADDITIONAL_FEATURES_TEXT)
+
+        # File uploading for features to be excluded
+        additional_features_file_buffer = st.file_uploader(
+            "Upload your CSV (comma(,) seperated) file here in which each row corresponds to an additional feature to be included for training.",
+            help="Upload your CSV (comma(,) seperated) file here in which each row corresponds to an additional feature to be included for training.",
+            type=["csv"],
         )
-        state["additional_features"] = multiselect(
-            "Select additional features for trainig:",
-            state.remainder,
-            default=None,
+        additional_features_df, add_df_warnings = load_data(
+            additional_features_file_buffer, "Comma (,)", header=None
         )
+        for warning in add_df_warnings:
+            st.warning(warning)
+
+        if len(additional_features_df) > 0:
+            st.markdown(
+                "The following additional features will be included for training:"
+            )
+            additional_features_df.columns = [
+                "Additional features to be included for training"
+            ]
+            st.table(additional_features_df)
+            additional_features_df_list = list(
+                additional_features_df.iloc[:, 0].unique()
+            )
+            state["additional_features"] = multiselect(
+                "Select additional features for training:",
+                help="Select additional features for training:",
+                options=state.remainder,
```

Comment:
Does this work if someone uploads entries that are not in the remainder list?

Reply:
Really nice point. Thanks a lot Max!

It was throwing error (we also missed this in `Exclude features`)
Now, I fixed both sides.

<img width="750" alt="Screenshot 2023-03-28 at 21 59 53" src="https://user-images.githubusercontent.com/49681382/228355766-50b209cd-6cd7-488e-960d-b03687320809.png">
<img width="741" alt="Screenshot 2023-03-28 at 22 00 41" src="https://user-images.githubusercontent.com/49681382/228355776-fba90860-b9b6-46f9-9317-e9ef57051246.png">
<img width="731" alt="Screenshot 2023-03-28 at 22 10 12" src="https://user-images.githubusercontent.com/49681382/228355786-59c5d3e4-ba57-4f89-b8a9-80ecaf6c0c58.png">